src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -262,10 +262,12 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
                     foreach (var subscriber in Subscribers)
                     {
-                        subscriber.Value.AddSubscriptionsAsync(newProjectContext);
+                        Shell.ThreadHelper.JoinableTaskFactory.Run(()=> subscriber.Value.AddSubscriptionsAsync(newProjectContext));
```

Comment:
Its unclear to me what the correct behavior is here.  This section of code is in a lock yet previously we would call `subscriber.Value.AddSubscriptionsAsync(newProjectContext)` and not observe the returned task or wait for it to complete. That seems to imply that we don't care about the lock, since we previously exited the lock statement without any guarantee that the async tasks had finished.  Here I am making the assumption that this was an error and we only want to exit the lock once all subscriptions have been added.